src/components/admin/private-route/PrivateRoute.test.tsx

```diff
@@ -0,0 +1,61 @@
+import { render, screen } from '@testing-library/react';
+import { useLocation } from 'react-router';
+import { useAdminContext } from '../../../context/admin-context-provider/AdminContextProvider';
+import { adminRoutes } from '../../../const/routers/routes';
+import { PrivateRoute } from './PrivateRoute';
+
+const mockUseAdminContext = useAdminContext as jest.MockedFunction<typeof useAdminContext>;
+const mockUseLocation = useLocation as jest.MockedFunction<typeof useLocation>;
+
+jest.mock('react-router', () => ({
+    Navigate: ({ to, state, replace }: any) => (
+        <div
+            data-testid="navigate"
+            data-to={to}
+            data-from={state.from.pathname}
+            data-replace={String(replace)}
+        />
+    ),
+    Outlet: () => <div data-testid="outlet" />,
+    useLocation: jest.fn(),
+}));
+jest.mock('../../../context/admin-context-provider/AdminContextProvider', () => ({
+    useAdminContext: jest.fn(),
+}));
+jest.mock('../../common/page-loader/PageLoader', () => ({
+    PageLoader: () => <div data-testid="loader" />,
+}));
+jest.mock('../../../const/routers/routes', () => ({
+    adminRoutes: { loginRoute: '/login' },
+}));
+
+describe('PrivateRoute', () => {
+    beforeEach(() => {
+        jest.clearAllMocks();
+    });
+
+    it('renders loader while loading', () => {
+        mockUseAdminContext.mockReturnValue({ isLoading: true, isAuthenticated: false } as any);
```

Comment:
Suggestion:
consider to use unknow rather then any
it's more propriet when you don't know what type will be.

Beause you still can assign value with any type but yuo can not do that with unknown type